libraries/AP_GPS/AP_GPS_NMEA.cpp

```diff
@@ -299,7 +299,38 @@ bool AP_GPS_NMEA::_term_complete()
         const bool crc_ok = _is_unicore? (_crc32 == crc) : (_parity == crc);
         if (crc_ok) {
             uint32_t now = AP_HAL::millis();
+
+#if AP_GPS_NMEA_SATELITES_INFO_ENABLED
+            int end;
+#endif
+
             switch (_sentence_type) {
+
+#if AP_GPS_NMEA_SATELITES_INFO_ENABLED
+            case _GPS_SENTENCE_GSV: {
+                end = 4;
+                if (_gsv.this_page_num == 0) {
+                    memset(state.satellites_prn,  0,  sizeof(state.satellites_prn));
+                    memset(state.satellites_used,  0,  sizeof(state.satellites_used));
+                    memset(state.satellites_snr,  0,  sizeof(state.satellites_snr));
+                    memset(state.satellites_elevation,  0,  sizeof(state.satellites_elevation));
+                    memset(state.satellites_azimuth,  0,  sizeof(state.satellites_azimuth));
+                }
+                if (_gsv.this_page_num == _gsv.all_page_num) {
+                    end = _gsv.tot_sv_visible - (_gsv.this_page_num - 1) * 4;
+                    state.satellites_visible = _gsv.num_gps + _gsv.num_glonass + _gsv.num_galileo + _gsv.num_baidou + _gsv.num_others;    
+                }
+                for (int y = 0; y < end; y++) {
+                    state.satellites_prn[y + (_gsv.this_page_num - 1) * 4]        = _gsv.svid[y];
```

Comment:
Looks like a buffer-overflow in the making here.

Create a variable to contain the offset, constrain when it is used to actually index the array.